pkg/cmd/secret/set/http.go

```diff
@@ -14,10 +15,10 @@ import (
 )
 
 type SecretPayload struct {
-	EncryptedValue string `json:"encrypted_value"`
-	Visibility     string `json:"visibility,omitempty"`
-	Repositories   []int  `json:"selected_repository_ids,omitempty"`
-	KeyID          string `json:"key_id"`
+	EncryptedValue string   `json:"encrypted_value"`
+	Visibility     string   `json:"visibility,omitempty"`
+	Repositories   []string `json:"selected_repository_ids,omitempty"`
```

Comment:
If the endpoint for setting Codespaces secrets accepts a different type for `selected_repository_ids` than other endpoints, then I would suggest defining a separate SecretPayload type to be used for Codespaces secrets only.

Reply:
Fixed in https://github.com/cli/cli/pull/4699/commits/7d0e2bc771c8de4b2fc523d5f9e32404f0ff787b